### ISSUE TITLE
Backfill missing and modified Littlepay customer_funding_source data

### ISSUE THREAD

**Issue by charlie-costanzo** [cal-itp/data-infra#3528]

**Comment by charlie-costanzo** [comment]
I don't think this will work, because the airflow operator will sync the most recent single file to our GCS bucket from the most recent partition in the Littlepay S3 bucket:
Airflow operator littlepay_raw_sync.py ([link](https://github.com/cal-itp/data-infra/blob/main/airflow/plugins/operators/littlepay_raw_sync.py)) syncs most recent data from Littlepay S3 buckets
This only takes a single file from the most recent partition, per get_latest_file ([link](https://github.com/cal-itp/data-infra/blob/a3e62e4d7967d3429d453bcc9011f8818b49c8d2/airflow/plugins/operators/littlepay_raw_sync.py#L107) to line in operator, [link](https://github.com/cal-itp/data-infra/blob/a3e62e4d7967d3429d453bcc9011f8818b49c8d2/packages/calitp-data-infra/calitp_data_infra/storage.py#L439) to source file)
But, will it work if all of the updated rows are in a single new file with our otherwise-expected new data?
We do handle new or updated data via hashing and deduplication
Example in stg_littlepay__customer_funding_source ([link](https://github.com/cal-itp/data-infra/blob/main/warehouse/models/staging/payments/littlepay/stg_littlepay__customer_funding_source.sql)): hashing [here](https://github.com/cal-itp/data-infra/blob/a3e62e4d7967d3429d453bcc9011f8818b49c8d2/warehouse/models/staging/payments/littlepay/stg_littlepay__customer_funding_source.sql#L28), deduplication using [this macro](https://github.com/cal-itp/data-infra/blob/a3e62e4d7967d3429d453bcc9011f8818b49c8d2/warehouse/macros/littlepay_staging_transforms.sql#L35) in the file [here](https://github.com/cal-itp/data-infra/blob/a3e62e4d7967d3429d453bcc9011f8818b49c8d2/warehouse/models/staging/payments/littlepay/stg_littlepay__customer_funding_source.sql#L53)
However, the hash above is for full row duplicates, so actually seems like it won't exclude the previous rows with missing data in favor of the new rows
BUT! I believe that this QUALIFY ROW_NUMBER()  partitioned by participant_id, funding_source_id, customer_id and ordered by littlepay_export_ts DESC  will ([link](https://github.com/cal-itp/data-infra/blob/a3e62e4d7967d3429d453bcc9011f8818b49c8d2/warehouse/models/staging/payments/littlepay/stg_littlepay__customer_funding_source.sql#L91))
On the join, new funding_source_id values will get picked up in fct_payment_rides_v2

tl;dr: it looks like we need to receive one file, but if we do, then the handling that I mentioned in the last bullet above should prioritize the new data and new principal_customer_ids via our QUALIFY statement

**Comment by charlie-costanzo** [comment]
Sent an email to Akos about this: 

The new data actually won't get picked up by our pipeline if it's sent as multiple files at the same time. So it seems like there are two options:
* See if everything can be published as a single file
* Publish the data as multiple files more than one hour apart. Our pipeline looks for new data every hour, and will only take the most recent file each hour. If we're able to publish the data files in a scattered way, they should be picked up
    Would it be possible to send a sample file ahead of the publishing of the modified historical data? We have assumptions about how the pipeline will handle the new rows, but would love to test them beforehand to see exactly how the data will be differentiated.
